### ISSUE TITLE
[Feat] Add flake8 tests as a new GH Action like in GRASS-core

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass-addons#371]
After the recent merge of  "PEP8: bulk fixes on entire grass7/ subdirectory tree with flake8" (#357) it might be a good idea to set up a new GitHub Action like https://github.com/OSGeo/grass/blob/master/.github/workflows/flake8.yml in core.

**Comment by lucadelu** [comment]
I try to run flake8 on a module but I got a lot of error mainly related to grass script format #%

```
flake8 --count --statistics --show-source --jobs=$(nproc)
./r.out.kde.py:17:1: E265 block comment should start with '# '
#%module
^
./r.out.kde.py:18:1: E265 block comment should start with '# '
#% description: Exports raster with variable transparency into an image file
^
./r.out.kde.py:19:1: E265 block comment should start with '# '
#% keyword: raster
^
./r.out.kde.py:20:1: E265 block comment should start with '# '
#% keyword: kernel density
^
./r.out.kde.py:21:1: E265 block comment should start with '# '
#% keyword: visualization
^
./r.out.kde.py:22:1: E265 block comment should start with '# '
#% keyword: transparency
^
./r.out.kde.py:23:1: E265 block comment should start with '# '
#% keyword: heatmap
^
./r.out.kde.py:24:1: E265 block comment should start with '# '
#%end
^
./r.out.kde.py:26:1: E265 block comment should start with '# '
#%option G_OPT_R_INPUT
^
./r.out.kde.py:27:1: E265 block comment should start with '# '
#% description: Raster map to be rendered with semi-transparency
^
./r.out.kde.py:28:1: E265 block comment should start with '# '
#%end
^
./r.out.kde.py:30:1: E265 block comment should start with '# '
#%option G_OPT_R_INPUT
^
./r.out.kde.py:31:1: E265 block comment should start with '# '
#% key: background
^
./r.out.kde.py:32:1: E265 block comment should start with '# '
#% description: Background raster map
^
./r.out.kde.py:33:1: E265 block comment should start with '# '
#%end
^
./r.out.kde.py:35:1: E265 block comment should start with '# '
#%option G_OPT_F_OUTPUT
^
./r.out.kde.py:36:1: E265 block comment should start with '# '
#% description: Rendered output file
^
./r.out.kde.py:37:1: E265 block comment should start with '# '
#%end
^
./r.out.kde.py:39:1: E265 block comment should start with '# '
#%option
^
./r.out.kde.py:40:1: E265 block comment should start with '# '
#% key: method
^
./r.out.kde.py:41:1: E265 block comment should start with '# '
#% type: string
^
./r.out.kde.py:42:1: E265 block comment should start with '# '
#% options: linear,logistic
^
./r.out.kde.py:43:1: E265 block comment should start with '# '
#% description: Method to scale transparency
^
./r.out.kde.py:44:1: E265 block comment should start with '# '
#%end
^
./r.out.kde.py:61:80: E501 line too long (97 > 79 characters)
    gscript.run_command('g.remove', name=','.join(TMPRAST), flags='f', type='raster', quiet=True)
                                                                               ^
24    E265 block comment should start with '# '
1     E501 line too long (97 > 79 characters)
25
```

I called flake8 (version 3.6.0) in the same way it is called in flake8.yml, @wenzeslaus do you have the same behavior in `scripts` directory?

**Comment by neteler** [comment]
@lucadelu be sure to have the GRASS GIS optimized config file `.flake8` file activated.

See this script for an example ([qa_run_flake8.zip](https://github.com/OSGeo/grass-addons/files/5746684/qa_run_flake8.zip)):

```
#!/bin/sh

# Markus Neteler, 2020
# script to recursively check GRASS GIS Python scripts for selected PEP8 violations
#
# Fedora: sudo pip3 install flake8 flake8-colors 
# Ubuntu: pip3 install flake8 flake8-colors

# fetch/update GRASS GIS optimized config
wget --quiet https://raw.githubusercontent.com/OSGeo/grass/master/scripts/.flake8 -O .flake8
echo 'format = ${cyan}%(path)s${reset}:${yellow_bold}%(row)d${reset}:${green_bold}%(col)d${reset}: ${red_bold}%(code)s${reset} %(text)s' >> .flake8

# run flake8 on directory tree
flake8 --count --statistics --show-source --jobs=4 .
rm -f .flake8
```

**Comment by lucadelu** [comment]
@neteler thanks the `.flake8` was the missing file.
Do we want to run a test for each sub-directory (raster, vector, etc) or just in the main directory?
However putting `.flake8` in the main directory is working also running flake8 command in the sub-directories

**Comment by neteler** [comment]
I would apply it to the entire `grass7/` subtree.

Running the script from above I still get a lot of errors but many may be fixed similar to #357

**Comment by lucadelu** [comment]
> I would apply it to the entire `grass7/` subtree.
> 

so in a single job? in [GRASS action](https://github.com/OSGeo/grass/blob/master/.github/workflows/flake8.yml) there are multiple jobs for different directories, I think it could be more clear and simple to read to have separated jobs, one for each directory 

> Running the script from above I still get a lot of errors but many may be fixed similar to #357

me too

> Ah, it contains some false alarms as well (to be masked or whatever).

ok, so we need to find which should be masked
